src/modules/sensors/vehicle_magnetometer/VehicleMagnetometer.cpp

```diff
@@ -141,18 +144,140 @@ void VehicleMagnetometer::ParametersUpdate(bool force)
 	}
 }
 
+void VehicleMagnetometer::MagCalibrationUpdate()
+{
+	// State variance assumed for magnetometer bias storage.
+	// This is a reference variance used to calculate the fraction of learned magnetometer bias that will be used to update the stored value.
+	// Smaller values will make the stored bias data adjust more slowly from flight to flight. Larger values will make it adjust faster.
+	static constexpr float magb_vref = 2.5e-7f;
+	static constexpr float min_var_allowed = magb_vref * 0.01f;
+	static constexpr float max_var_allowed = magb_vref * 100.f;
+
+	if (_armed && !_landed) {
+		for (int i = 0; i < _estimator_sensor_bias_subs.size(); i++) {
+			estimator_sensor_bias_s estimator_sensor_bias;
+
+			if (_estimator_sensor_bias_subs[i].update(&estimator_sensor_bias)
+			    && (hrt_elapsed_time(&estimator_sensor_bias.timestamp) < 1_s)) {
+
+				const Vector3f bias{estimator_sensor_bias.mag_bias};
+				const Vector3f bias_variance{estimator_sensor_bias.mag_bias_variance};
+
+				const bool valid = (estimator_sensor_bias.mag_device_id != 0)
+						   && estimator_sensor_bias.mag_bias_valid
+						   && (bias_variance.min() > min_var_allowed)
+						   && (bias_variance.max() < max_var_allowed);
+
+				_mag_cal_valid[i].set_state_and_update(valid, estimator_sensor_bias.timestamp);
+
+				if (_mag_cal_valid[i].get_state()) {
+					// find corresponding mag calibration
+					for (int mag_index = 0; mag_index < MAX_SENSOR_COUNT; mag_index++) {
+						if (_calibration[mag_index].device_id() == estimator_sensor_bias.mag_device_id) {
+
+							const Vector3f mag_offset{_calibration[mag_index].BiasCorrectedSensorOffset(bias)};
+
+							// store best state to save back to parameters later, averaged across valid estimates
+							if (_mag_cal_available[mag_index]) {
+								_mag_cal_offset[mag_index] = (_mag_cal_offset[mag_index] + mag_offset) * 0.5f;
```

Comment:
I can't find where this average of the two last valid estimates comes from. It is probably unnecessary given the way that the weighting is applied on line 221

Reply:
Is this being used to accomodate updates from more than one EKF using the same magnetometer?